webapp/app/.server/service/l2-cancellations.ts

```diff
@@ -122,16 +147,11 @@ async function fetchProposalsFromL2Governor(type: L2CancellationType, from: bigi
   );
 
   const states = await Promise.all(
-    activeProposals.map((p) =>
-      l2Rpc.contractRead(getL2GovernorAddress(type), "state", ZK_GOV_OPS_GOVERNOR_ABI, z.number(), [
-        p.proposalId,
-      ])
-    )
+    activeProposals.map((p) => getL2ProposalState(p.type, p.proposalId))
   );
 
-  return activeProposals.filter(
-    (_, i) =>
-      states[i] === L2_CANCELATION_STATES.Active || states[i] === L2_CANCELATION_STATES.Pending
+  return activeProposals.filter((_, i) =>
+    VALID_CANCELLATION_STATES.some((state) => state === states[i])
```

Comment:
```suggestion
    VALID_CANCELLATION_STATES.includes(states[i])
```

Reply:
Nice find! Actually, this is a tricky one. That doesn't work because state[i] might be undefined. That's why I made function to abstract that away that then I forgot to use here. Changed